src/index.ts

```diff
@@ -407,17 +407,23 @@ export const plugin = (opts?: { debug: boolean }): Plugin => {
     async transform(_code, id) {
       if (!id.endsWith('.elm')) return
       compilableFiles.delete(id)
+
       const isBuild = process.env.NODE_ENV === 'production'
+      const dependencies: string[] = await compiler.findAllDependencies(id)
+      compilableFiles.set(id, new Set(dependencies))
+
       try {
         const compiled = await compiler.compileToString([id], {
           output: '.js',
           optimize: !debug && isBuild,
           verbose: isBuild,
           debug,
         })
-        const dependencies: string[] = await compiler.findAllDependencies(id)
-        dependencies.forEach(this.addWatchFile)
-        compilableFiles.set(id, new Set(dependencies))
+
+        if (this.addWatchFile) {
```

Comment:
I'm not sure why this is `undefined`

Reply:
hmm, how can we reproduce this situation. as long as I look through `addWatchFile` exists on context always